### PR TITLE
Fixes #11245: Add constraint on hash_type

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/11245

### CODE REVIEW

File: tree/30_generic_methods/file_from_shared_folder.cf [hunk 1]
```diff
@@ -21,7 +21,8 @@
 #
 # @parameter source      Source file (path relative to Rudder shared folder)
 # @parameter destination Destination file (absolute path on the target node)
-# @parameter hash_type   Hash algorithm used to check if file is updated
+# @parameter hash_type   Hash algorithm used to check if file is updated (md5, sha1, sha256)
```
Comment (peckpeck) [inline tree/30_generic_methods/file_from_shared_folder.cf]: You should document that this parameter is always treated as md5 on cfengine.

**Comment by VinceMacBuche** [comment]
Commit modified

File: tree/30_generic_methods/file_from_shared_folder.cf [hunk 2]
```diff
@@ -21,7 +21,8 @@
 #
 # @parameter source      Source file (path relative to Rudder shared folder)
 # @parameter destination Destination file (absolute path on the target node)
-# @parameter hash_type   Hash algorithm used to check if file is updated
+# @parameter hash_type   Hash algorithm used to check if file is updated (md5, sha1, sha256), only md5 can be used on cfengine agent
```
Comment (ncharles) [inline tree/30_generic_methods/file_from_shared_folder.cf]: it's not used on cfengine, so why not simply state that it is not used for cfengine agent ?

Comment (VinceMacBuche) [inline tree/30_generic_methods/file_from_shared_folder.cf]: I will write that on cfengine it will always be treated as md5 for now

Comment (ncharles) [reply tree/30_generic_methods/file_from_shared_folder.cf]: ping ?

**Comment by VinceMacBuche** [comment]
Commit modified

**Pull request merged**